src/verifier.rs

```diff
@@ -78,109 +78,104 @@ pub enum VerifierError {
     #[error("Invalid register specified at instruction {0}")]
     InvalidRegister(usize),
 }
-impl UserDefinedError for VerifierError {}
-impl From<VerifierError> for UserError {
-    fn from(error: VerifierError) -> Self {
-        UserError::VerifierError(error)
-    }
+
+fn adj_insn_ptr(insn_ptr: usize) -> usize {
+    insn_ptr + ebpf::ELF_INSN_DUMP_OFFSET
 }
 
 fn check_prog_len(prog: &[u8]) -> Result<(), VerifierError> {
     if prog.len() % ebpf::INSN_SIZE != 0 {
         return Err(VerifierError::ProgramLengthNotMultiple);
     }
-    if prog.len() > ebpf::PROG_MAX_SIZE {
```

Comment:
How about keeping `PROG_MAX_SIZE` as a fallback, but increasing it to match the max account size?

Reply:
Why do we care about enforcing a program max in the verifier?  Sounds like two places that can get out of sync.  If anything the loader itself should probably enforce program account size rather than an elf verifier